docs/MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,212 @@
+
+### babel6 is now babel
+
+The `@webpack-blocks/babel6` package has been renamed to `@webpack-blocks/babel`. You can now choose which version of babel to use, since `babel-core` is now a peer dependency (see below).
+
+### babel-core as a peer dependency
+
+`babel-core` is now a peer dependency of the `babel` block, so you can control which version of `babel-core` to use. Make sure to `npm install --save-dev babel-core` in your application.
+
+### `webpack` no longer exported
+
+The `webpack` instance is no longer exported by `@webpack-blocks/webpack`. Use `const webpack = require('webpack')` instead which is more explicit.
+
+### `resolve()` instead of `resolveAliases()`
+
+The `resolveAliases` utility in `@webpack-blocks/webpack` has been replaced by the more generic `resolve`.
+
+```diff
+module.exports = createConfig([
+-  resolveAliases({ package: path.resolve('./src/package') })
++  resolve({
++    alias: { package: path.resolve('./src/package') }
++  })
+])
+```
+
+### Removed `devServer.reactHot()`
+
+The `devServer.reactHot()` block has never worked too well. We also decided not to ship framework-specific blocks as core features.
+
+If you feel like this is a feature you really want to have, create a new fancy React hot loader block and share it with the world!
+
+### Requiring node 6+ now
+
+Please use node 6 or newer. We don't transpile the webpack-blocks code for legacy node versions anymore.
+
+---
+
+## Additional features
+
+### webpack v3
+
+We officially support webpack 3 now 🎉
+
+### `@webpack-blocks/uglify` package
+
+The `uglify` package is a convenience block for easily configuring JS minification using the [uglifyjs-webpack-plugin](https://www.npmjs.com/package/uglifyjs-webpack-plugin) which also supports minifying untranspiled ES2015 code.
+
+See the package's README for further details.
+
+### `setEnv()`
+
+New utility block in `@webpack-blocks/webpack`. Replaces constants in your source code with a values from `process.env` using the [webpack.EnvironmentPlugin](https://webpack.js.org/plugins/environment-plugin/).
+
+Using `setEnv` multiple times results in a single EnvironmentPlugin instance configured to do all the replacements.
+
+```js
+module.exports = createConfig([
+  setEnv(['NODE_ENV']),
+  setEnv({
+    BABEL_ENV: 'development', // use 'development' unless process.env.BABEL_ENV is defined
+    PORT: 3000,
+  }),
+])
+```
+
+### `resolve()` will prepend custom `extensions` (instead of appending them)
+
+The new `resolve` utility in `@webpack-blocks/webpack` will prepend custom `extensions` you pass to it, rather than appending them.
+
+This way extensions you set later in your `createConfig()` will have precedence over extensions you set earlier (webpack tries to match against the `extensions` from the first to last).
+
+---
+
+## Migrating blocks
+
+### A concise migration example
+
+A simple babel block in the old v0.4:
```

Comment:
Probably not the best example — it feels like I have to do something with the default `babel` block.

Reply:
Yeah, but it's easy to understand and nearly everyone knows it. Couldn't think of a better example.

Reply:
I think there might not be enough context for people to understand what it is for. There should be an introduction paragraph like `In version 1.0 we are introducing a new API for writing blocks`, and also probably a few words to answer a question 'Why?' (to conform with the new block usage, to make an expected effect of blocks more explicit, ...).

Then I'd use something like
```
A simple Babel block written for the old v0.4 API would look like this:
```
And later
```
The same block updated for the v1.0 API:
```

Reply:
I added an introductary paragraph to the top of the migration guide. Does that work for you or should I change it?

Also updated the captions for the sample babel block.